internal/app/referenceserver/impl.go

```diff
@@ -55,14 +56,10 @@ func (s *conformanceServer) Unary(
 		ctx,
 		req.Msg.ResponseDefinition,
 		req.Header(),
+		nil,
 		[]*anypb.Any{msgAsAny},
 	)
-	// If connectErr is not nil, then the response definition must not be nil.
-	// Read the headers and trailers from the definition and add them to connectErr
-	// and return
 	if connectErr != nil {
-		internal.AddHeaders(req.Msg.ResponseDefinition.GetResponseHeaders(), connectErr.Meta())
-		internal.AddHeaders(req.Msg.ResponseDefinition.GetResponseTrailers(), connectErr.Meta())
```

Comment:
Why remove these?

Reply:
I didn't actually remove these, I just moved them to be added in `parseUnaryDefinition`.

Reply:
Yeah, my bad. I did see that and thought I removed this draft comment.